db/constraints.c

```diff
@@ -2391,8 +2391,8 @@ int update_constraint_genid(struct ireq *iq, int opcode, int blkpos, int flags,
                 goto done;
             }
 
-            rc = insert_add_op(iq, opcode, rrn, -1, new_genid, ins_keys, blkpos,
-                               flags);
+            rc = insert_add_op(iq, opcode, rrn, curop->ixnum, new_genid, curop->ins_keys, curop->blkpos,
+                               curop->flags);
```

Comment:
The `flags` argument also has UPDFLAGS_CASCADE set. We'll have to check the effects of loosing that bit. Otherwise, the all new additions look good. Also, curious, what is the underlying issue that you're trying to fix?

Reply:
So the bug I was investigating is in fact with a txn where a row gets updated twice from the same txn (once in cascade): https://github.com/bloomberg/comdb2/pull/2763
I modified this pr to just modify the record in place and this change fixes that bug.